BirthdayGreetingsTest/addNewBirthdayFriendToDB.js

```diff
@@ -1,40 +1,39 @@
 import sqlite3 from "sqlite3";
 import { convertDate } from "./convertDate.js";
 
-export function addNewBirthdayFriendToDB(
+export async function addNewBirthdayFriendToDB(
   lastName,
   firstName,
   dateOfBirth,
   email
 ) {
-  // Validate the input
-  if (!lastName || !firstName || !email || !dateOfBirth) {
-    throw new Error("Missing required fields");
-  }
-
-  dateOfBirth = convertDate(dateOfBirth); // Convert the date to the correct format
+  try {
+    // Validate the input
+    if (!lastName || !firstName || !email || !dateOfBirth) {
+      throw new Error("Missing required fields");
+    }
 
-  if (!/^[a-zA-Z]+$/.test(lastName) || !/^[a-zA-Z]+$/.test(firstName)) {
-    throw new Error("Name fields should only contain letters");
-  } else if (lastName.length > 20) {
-    try {
-      lastName = lastName.slice(0, 20);
-      console.log(`Last name truncated to ${lastName} as it was too long`);
-    } catch (error) {
-      throw new Error("Error truncating last name");
+    if (!/^[a-zA-Z]+$/.test(lastName) || !/^[a-zA-Z]+$/.test(firstName)) {
+      throw new Error("Name fields should only contain letters");
+    } else if (lastName.length > 20 || firstName.length > 20) {
+      throw new Error("Name fields should not be longer than 20 characters");
     }
-  } else if (firstName.length > 20) {
-    try {
-      firstName = firstName.slice(0, 20);
-      console.log(`First name truncated to ${firstName} as it was too long`);
-    } catch (error) {
-      throw new Error("Error truncating first name");
+
+    if (!/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(email)) {
```

Comment:
```suggestion
    if (invalidEmail(email)) {
```
Something like this would be better for the reader.